src/lib/fixer.ts

```diff
@@ -131,6 +138,164 @@ export class _PmdFixGenerator extends FixGenerator {
             title: 'Clear Single Diagnostic',
             arguments: [this.document.uri, this.diagnostic]
         };
+
         return action;
     }
+
+    public generateClassLevelSuppression(): vscode.CodeAction {
+        // Find the end-of-line position of the class declaration where the diagnostic is found.
+        const classStartPosition = this.findClassStartPosition(this.diagnostic, this.document);
+        // const classEndOfLinePosition = 0;
+
+        const action = new vscode.CodeAction(messages.fixer.supressOnClass, vscode.CodeActionKind.QuickFix);
+        action.edit = new vscode.WorkspaceEdit();
+    
+        // Determine the appropriate suppression rule based on the type of diagnostic.code
+        let suppressionRule: string;
+        if (typeof this.diagnostic.code === 'object' && this.diagnostic.code !== null && 'value' in this.diagnostic.code) {
+            suppressionRule = `PMD.${this.diagnostic.code.value}`;
+        } else {
+            suppressionRule = `PMD`;
+        }
+    
+        // Extract text from the start to end of the class declaration to search for existing suppressions
+        const classText = this.findLineBeforeClassStartDeclaration(classStartPosition, this.document);
+        const suppressionMatch = classText.match(this.suppressionRegex);
+    
+        if (suppressionMatch) {
+            // If @SuppressWarnings exists, check if the rule is already present
+            const existingRules = suppressionMatch[1].split(',').map(rule => rule.trim());
+            if (!existingRules.includes(suppressionRule)) {
```

Comment:
Just wondering about case sensitivity of the rules. If they are case insensitive, then we probably should add to the map a .toLowerCase() and then add it to the suppressionRule when checking for existence.

Reply:
Follow-up question: Is it an error case if the rule is already present, or is it valid and nothing should happen?

Reply:
I have added case insensitivity in the regex and added tests. It is an error case when a rule is already present and violations are present. One scenario I can think of is when user ran the scanner without the rule and manually added the rule and then tried to run the suppress at the class level. I'm not handling that part.

Reply:
Adding case insensitivity in the regex doesn't help with this line does it existingRules.includes(suppressionRule)) ?

I'd recommend something like
            const existingRules = suppressionMatch[1].split(',').map(rule => rule.trim().toLowerCase());
            if (!existingRules.includes(suppressionRule.toLowerCase())) {

Reply:
As discussed offline, this has an unintended side effect. It changes all existing rules to lower case since we concatenate the existing rules (which were made to be lower case for comparison) to new rules. So, we decided to leave this as is.